### PR TITLE
chains: add api chain

### PR DESCRIPTION
**Pull request by byebyebruce**
Add api chain

### PR Checklist

- [x] Read the [Contributing documentation](https://github.com/tmc/langchaingo/blob/main/CONTRIBUTING.md).
- [x] Read the [Code of conduct documentation](https://github.com/tmc/langchaingo/blob/main/CODE_OF_CONDUCT.md).
- [x] Name your Pull Request title clearly, concisely, and prefixed with the name of primarily affected package you changed according to [Good commit messages](https://go.dev/doc/contribute#commit_messages) (such as `memory: add interfaces for X, Y` or `util: add whizzbang helpers`).
- [x] Check that there isn't already a PR that solves the problem the same way to avoid creating a duplicate.
- [x] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `fixes #123`).
- [x] Describes source of new concepts.
- [x] References existing implementations as appropriate.
- [x] Contains test coverage for new functions.
- [x] Passes all [`golangci-lint`](https://golangci-lint.run/) checks.

### CODE REVIEW

File: chains/api_test.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+		"question": "What is the weather of [latitude:52.52, longitude:13.419998]?",
+	}
+	result, err := Call(context.Background(), chain, q)
+	require.NoError(t, err)
+
+	answer, ok := result["answer"].(string)
+	if !ok {
+		t.Fatal("expected answer to be a string")
+	}
+	require.True(t, strings.Contains(answer, "weather"), "expected 58.708 in result")
```
Comment (tmc) [inline chains/api_test.go:68]: this seems awfully specific.

I'd expect this test to do something like start a temporary test

File: chains/api.go [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+	reqP := prompts.NewPromptTemplate(_llmAPIURLPrompt, []string{"api_docs", "question"})
+	reqC := NewLLMChain(llm, reqP)
+
+	respP := prompts.NewPromptTemplate(_llmAPIResponsePrompt, []string{"api_docs", "question", "api_url", "api_response"})
+	respC := NewLLMChain(llm, respP)
+
+	return APIChain{
+		RequestChain: reqC,
+		AnswerChain:  respC,
+		Requester:    http.DefaultClient,
```
Comment (tmc) [inline chains/api.go:60]: typically we should optionally allow specification of the http client

File: chains/api.go [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+}
+
+// APIChain is a chain used for request api.
+type APIChain struct {
+	RequestChain *LLMChain
+	AnswerChain  *LLMChain
+	Requester    HTTPRequester
+}
+
+func NewAPIChain(llm llms.LLM) APIChain {
```
Comment (tmc) [inline chains/api.go:50]: merits a comment

Comment (byebyebruce) [reply chains/api.go]: Updated.

**Review by tmc — APPROVED** [review]
LGTM

**Pull request merged**